### ISSUE TITLE
`zig cc` can't accept -- in args

### ISSUE THREAD

**Issue by iacore** [ziglang/zig#14837]
### Zig Version

0.11.0-dev.1898+36d47dd19

### Steps to Reproduce and Observed Behavior

```
> echo "int main() {}" > program.c

> zig cc -- program.c
error: Unknown Clang option: '--'

> clang -- program.c
# success
```

### Expected Behavior

It should parse like clang. args after `--` are not flags.

**Comment by yujiri8** [comment]
Are you sure that's actually how clang works? I ran `clang -- -program.c` where `-program.c` exists and it said:
```
error: unknown argument: '-program.c'
```

**Comment by locriacyber** [comment]
`-program.c` should be a file name.

**Comment by yujiri8** [comment]
Yes, clang for me shows that error message when `-program.c` is the name of a C source file. My clang version is 15.0.7, what version are you on?

**Comment by locriacyber** [comment]
> Yes, clang for me shows that error message when `-program.c` is the name of a C source file. My clang version is 15.0.7, what version are you on?
I gave a bad example. What about `clang -- program.c`

**Comment by yujiri8** [comment]
Strange, I've found that actually clang for me does process the `--` as expected, even with `-program.c`... but only if `-program.c` *doesn't* exist.

`clang -- -program.c` where `-program.c` doesn't exist:
```
clang-15: error: no such file or directory: '-program.c'
clang-15: error: no input files
```
`clang -- -program.c` where it does exist:
```
error: unknown argument: '-program.c'
```

**Comment by locriacyber** [comment]
Did we find a bug in clang?

**Comment by yujiri8** [comment]
I dunno, seems like it :shrug:

**Comment by andrewrk** [comment]
This is not demonstrated to be a bug yet. Please improve the bug report

**Comment by iacore** [comment]
> This is not demonstrated to be a bug yet. Please improve the bug report

Updated

**Comment by hadeutscher** [comment]
Hey,

Why doesn't zigcc accept the `--` delimiter?

While this isn't *technically* a bug, it does cause quite a lot of trouble for the rest of the community - for example, the Rust `cc` crate has special-case handling for the zig compiler which removes the `--` delimiter, which breaks every once in a while, [such as yesterday](https://github.com/rust-lang/cc-rs/issues/1360), and [wreaks havoc](https://github.com/hadeutscher/RustyRail/actions/runs/12726266024/job/35495196554) on projects built using the `dist` framework (which uses zigcc internally for cross-compiling aarch64 targets).

Also, clang accepts this delimiter, so if zigcc aims to be a drop-in replacement for clang, it really is a bug.

Seeing as you already [wrote the pull request](https://github.com/ziglang/zig/pull/14862) that fixes this, maybe we can just get that one merged and be over with this?

Thanks
